test/integration/features/profile.feature

```diff
@@ -2,104 +2,161 @@
-      """
-      - foo		Running		(Active)
-      - minishift	Running
-      """
+  Scenario Outline: As user, I can create profile with alphanumeric character including '_' and '-' in profile name
+    When executing "minishift profile set <profilename>" succeeds
+     Then profile <profilename> should have state "Does Not Exist"
+     And profile <profilename> should be the active profile
 
-  Scenario: User should be able set 'minishift' as the active profile
-      When executing "minishift profile set minishift" succeeds
-      Then stdout should contain
-      """
-      Profile 'minishift' set as active profile
-      """
+  Examples: Correct profile names
+    | profilename |
+    | Test-123    |
+    | profile_    |
+    | P1_name-    |
+    | vm_45       |
+    | 20          |
+    | random001z  |
+    | 00_XYZ-     |
+    | test_Pro-01 |
+    | foo         |
 
-Scenario: User should be able to list 'minishift' as the active profile
-    When executing "minishift profile list" succeeds
-    Then stdout should contain
-    """
-    - minishift	Running		(Active)
-    """
+  Scenario: As user, I can switch between existing profiles
+    When executing "minishift profile set Test-123" succeeds
+     Then profile "Test-123" should be the active profile
+     And executing "minishift profile set minishift" succeeds
+     Then profile "minishift" should be the active profile
 
-Scenario: User should be able to delete profile 'foo'
-    When executing "minishift profile delete foo --force" succeeds
-    Then stdout should contain
-    """
-    Profile 'foo' deleted successfully
-    """
+  Scenario: As user, I can start Minishift with default profile 'minishift'
+    Given profile "minishift" has state "Does Not Exist"
+    And profile "minishift" is the active profile
+     When executing "minishift start" succeeds
+     Then profile "minishift" should have state "Running"
+     When executing "minishift profile list" succeeds
+     Then stdout should match "-\s*minishift\s*Running\s*\(Active\)"
 
-Scenario: User can not delete default profile 'minishift'
-    When executing "minishift profile delete minishift --force"
-    Then exitcode should equal "1"
-     And stderr should contain
+  Scenario: As user, I can apply independent settings in profile 'foo'
+    When executing "minishift profile set foo" succeeds
+     Then profile "foo" should be the active profile
+     When executing "minishift --profile minishift config get memory" succeeds
+     Then stdout should match "<nil>"
+     And executing "minishift --profile minishift addons list" succeeds
+     And stdout should match "registry-route\s*: disabled\s*P\(0\)"
+     And executing "minishift addons enable registry-route" succeeds
+     And executing "minishift config set memory 5120" succeeds
+     And executing "minishift config get memory" succeeds
+     And stdout should match "5120"
+     When executing "minishift start" succeeds
+     Then stdout should contain
```

Comment:
This test is little confusing to me. In the scenario we are saying that it is about `apply independent settings in profile 'foo'` but in the test we are applying settings to both `minishift` and `foo` and we are verifying the settings of memory for profile `foo`. Do we really need to set config for `minishift`.

Reply:
yes, we need to set the atleast some config setting in profile ```foo``` to verify that it should not be reflected in default profile ```minishift```.

> but in the test we are applying settings to both minishift and foo

No. We are applying setting only for foo and verifying with the default setting of minishift.

Reply:
@LalatenduMohanty, you need to do something. The scenario is that you can do something *independently*. If you just modify settings for profile 'foo' and assert that they apply how can you make sure that these changes are not applied another existing profile.  In this case you are not asserting your actual scenario. As @amitkrout is saying you need to make sure that something else (eg the default mininishift profile) stays untouched.